src/plugins/intel_gpu/src/graph/impls/registry/gemm_impls.cpp

```diff
@@ -19,7 +19,10 @@ const std::vector<std::shared_ptr<cldnn::ImplementationManager>>& Registry<gemm>
     static const std::vector<std::shared_ptr<ImplementationManager>> impls = {
         OV_GPU_CREATE_INSTANCE_ONEDNN(onednn::GemmImplementationManager, shape_types::static_shape)
         OV_GPU_GET_INSTANCE_OCL(gemm, shape_types::static_shape)
-        OV_GPU_GET_INSTANCE_OCL(gemm, shape_types::dynamic_shape)
+        OV_GPU_GET_INSTANCE_OCL(gemm, shape_types::dynamic_shape,
+            [](const program_node& node) {
+                return !node.can_use(impl_types::onednn);
```

Comment:
Does oneDNN support efficient kernels caching for gemm? If not, this could cause runtime kernel recompilation and drop performance in some cases. This change probably requires wider performance check

Reply:
Checked on LNL with qwen2 and llama3, no performance drops were detected (with and w/o sdpa)